doc/programming/parsing.rst

```diff
@@ -1654,3 +1654,112 @@ control Spicy's gap handling, including when to stop buffering data
 because you know nothing further will arrive anymore. Spicy can also
 notify you about unsuccessful reassembly through a series of built-in unit hooks.
 See :ref:`type_sink` for a reference of the available functionality.
+
+
+.. _unit_context:
+
+Contexts
+========
+
+Parsing may need to retain state beyond any specific unit's lifetime.
+For example, a UDP protocol may want to remember information across
+individual packets (and hence units), or a bi-directional protocol may
+need to correlate the request side with the response side. One option
+for implementing this is maintaining the state in :ref:`global
+variables <variables>`. However, doing so can be cumbersome because a
+global will often not match semantics well. As an alternative, a unit
+can make use of a dedicated *context* value, which is an instance of a
+custom type that has its lifetime determined suitably by the host
```

Comment:
We talked before how this allows creating new reference cycles. Looking at this patch, it seems that this is still possibility for our users of contexts. If I read that right, let's capture it in a improvement ticket.

Reply:
as we chatted about, it's a general problem we have without a cycle detector. As this point, context at least don't make it worst I believe. :)